grades/tasks.py

```diff
@@ -0,0 +1,127 @@
+"""
+Tasks for the course app
+"""
+import logging
+
+from celery import group
+from celery.result import GroupResult
+from django.core.cache import cache
+
+from courses.models import CourseRun
+from grades import api
+from grades.models import FinalGradeRunInfo
+from micromasters.celery import async
+from micromasters.utils import chunks
+
+
+CACHE_ID_BASE_STR = "freeze_grade_{0}"
+
+log = logging.getLogger(__name__)
+
+
+@async.task
+def find_course_runs_for_grade_freeze():
+    """
+    Async task that takes care of finding all the course
+    runs that can freeze the final grade to their students.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+    # pull all the course run already frozen
+    frozen_runs = FinalGradeRunInfo.get_frozen_course_runs()
+    runs_to_freeze = CourseRun.get_runs_to_freeze(exclude_list=frozen_runs)
+    for run in runs_to_freeze:
+        freeze_course_run_final_grades.delay(run)
+
+
+@async.task
+def freeze_course_run_final_grades(course_run):
+    """
+    Async task manager to freeze all the users' final grade in a course run
+
+    Args:
+        course_run (CourseRun): a course run model object
+
+    Returns:
+        None
+    """
+    # no need to do anything if the course run is not ready
+    if not course_run.can_freeze_grades:
+        log.info('the grades course "%s" cannot be frozen yet', course_run.edx_course_key)
+        return
```

Comment:
same comment as above with respect to this log line. will we get anything useful from this in our logs? also, this check is repeated in `grades.api.freeze_user_final_grade` and here. is it necessary to have them both?